test/services/view/HolderOfAuthorisationCodesSpec.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.view
+
+import java.util.Locale.ENGLISH
+
+import base.UnitSpec
+import config.AppConfig
+import connectors.FileBasedCodeListConnector
+import org.mockito.Mockito.{reset, when}
+import org.scalatest.BeforeAndAfterEach
+
+class HolderOfAuthorisationCodesSpec extends UnitSpec with BeforeAndAfterEach {
+
+  private val appConfig = mock[AppConfig]
+
+  private lazy val codeListConnector = new FileBasedCodeListConnector(appConfig)
+
+  private lazy val holderOfAuthorisationCodes = new HolderOfAuthorisationCodes(codeListConnector)
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+
+    reset(appConfig)
+    when(appConfig.holderOfAuthorisationCodes).thenReturn("/code-lists/holderOfAuthorisationCodes.json")
+    when(appConfig.procedureCodesListFile).thenReturn("/code-lists/manyCodes.json")
```

Comment:
Why am I forced here to also provide the config for "Procedure codes" and "Additional Procedure codes" when what I want is just to test in isolation "Holder of Authorisation codes" ?

Reply:
Well if you mocked FileBasedCodeListConnector rather than use the real connector class you could test this service class in isolation. It is the same deal as we have with the AppConfigSpec. As we want them to fail fast then we must provide them with all their requirements when harnessing them for tests.

Reply:
With "in isolation" I meant not coupled with the other "codes", not testing `HolderOfAuthorisationCodesSpec` in isolation. In this test I want to use the real connector because I want to test if the "Holder of Authorisation" codes are loaded ordered as per requirements, and not using fake but the real values. AppConfigSpec is not useful at all all from this point of view because it is only using fake values.

Reply:
I'd suggest not using the real codes file to test that the order is maintained, otherwise if the real codes file changes order in the future the tests will break (even though the code is actually still fine).

Reply:
But that's what I want, I want the test fails if the actual result is different from what is expected from the loader. Especially with the current file-based implementation, which in case of changes will however require a new deployment. With a future, if it will ever happen :smirk: , network-based implementation the real values will be actually not so relevant any more.